sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -191,11 +193,15 @@ case class SortArray(base: Expression, ascendingOrder: Expression)
 }
 
 /**
- * Checks if the array (left) has the element (right)
+ * Checks if the array (left) has the element (right) and pattern match in
+ * case left is Array of type string
  */
+
 @ExpressionDescription(
-  usage = "_FUNC_(array, value) - Returns TRUE if the array contains the value.",
-  extended = " > SELECT _FUNC_(array(1, 2, 3), 2);\n true")
+  usage = """_FUNC_(array, value) - Returns TRUE if the array contains the value or
```

Comment:
The problem is that this changes the behavior of this method and even makes it a little surprising. Before, `..array("Mr.X"), "Mrox"..` didn't match but now it does. Accepting strings and regexes in the same place is inherently ambiguous. I don't know if we'd change the meaning of an existing function like this.